src/kbmod/analysis/wcs_utils.py

```diff
@@ -0,0 +1,143 @@
+    image_fov : astropy.units.Quantity, optional
+        The field of view of the larger image dimension, by default None
+        If provided the pixel scale is calculated from the image shape and the field of view
+    """
+    if min(img_shape) < 1:
+        raise ValueError("The image shape must be greater than zero.")
+    if ref_val is None:
+        ref_val = astropy.coordinates.SkyCoord(
+            ra=0 * astropy.units.deg, dec=-40 * astropy.units.deg, frame="icrs"
+        )
+    if ref_pix is None:
+        # this is the central one based pixel coordinate
+        ref_pix = [((img_shape[0] - 1) / 2 + 1), ((img_shape[0] - 1) / 2 + 1)]
+    _solve_for_image_fov = False
+    if pixel_scale is None:
+        if image_fov is None:
+            pixel_scale = 0.2 * astropy.units.arcsec / astropy.units.pix
+        else:
+            # not convinced this is correct. It does not seem to result in an image fov of image_fov
+            # but then I could be measuring the image fov incorrectly. The tangent performed no better.
+            pixel_scale = image_fov / max(img_shape) / astropy.units.pix
+            _solve_for_image_fov = solve_for_image_fov
+    # Define the WCS transformation
+    wcstan = _wcs_tangent_projection(ref_val, img_shape, ref_pix, pixel_scale)
+    if _solve_for_image_fov:
+        wcstan = solve_wcs_for_fov(wcstan, image_fov)
+    return wcstan
+
+
+def _wcs_tangent_projection(ref_val, img_shape, ref_pix, pixel_scale):
+    """Construct a WCS tangent projection.
+    Similar to construct_wcs_tangent_projection but without the checks, default values, and image_fov parameter.
+    """
+    pixel_scale_value = pixel_scale.to(astropy.units.deg / astropy.units.pix).value
+    wcstan = astropy.wcs.WCS(naxis=2)
+    wcstan.array_shape = img_shape
+    wcstan.wcs.crpix = ref_pix
+    wcstan.wcs.cdelt = [-pixel_scale_value, pixel_scale_value]
+    wcstan.wcs.ctype = ["RA---TAN", "DEC--TAN"]
+    wcstan.wcs.crval = [ref_val.ra.deg, ref_val.dec.deg]
+    return wcstan
+
+
+def solve_wcs_for_fov(wcs, fov):
+    """Solve for the pixel scale of the WCS given the field of view."""
+    # Assuming that the pixel scale is the arc of the reference pixel.
+    ref_pixel = wcs.wcs.crpix
+    image_size = wcs.array_shape
+    pixel_scale_mark = wcs.wcs.cdelt[1]
+    tolerance = 1e-12
+    iterations = 1000
+    pixel_scale_bounds = [pixel_scale_mark, pixel_scale_mark]
+    refsep2 = _update_calc_fov(wcs, ref_pixel, image_size, pixel_scale_mark)
+    delta = pixel_scale_mark / 16.0
+    while 0 < iterations and tolerance < abs(refsep2[0] - fov).value and fov <= refsep2[0]:
+        delta *= 2.0
+        pixel_scale_bounds[0] -= delta
+        refsep2 = _update_calc_fov(wcs, ref_pixel, image_size, pixel_scale_bounds[0])
+        iterations -= 1
+    delta = pixel_scale_mark / 16.0
+    while 0 < iterations and tolerance < abs(refsep2[0] - fov).value and refsep2[0] < fov:
```

Comment:
Do you need to reset `iterations` between the loops? Or is `iterations` meant to be the total iterations over all three loops?

Reply:
It is meant as a all else fails kill the loop. I know it is ugly but then the entire guess and check for this exists only because I did not figure out how to set the pixel scale to achieve the image field of view correctly. And that is what I wanted.